File: Interview Prep/Find the maximum product of 4 numbers in an array /MaxProductOfFour.java
import java.util.Arrays;
import java.util.Scanner;

public class MaxProductOfFour {
  public static long maxProductOfFour(int[] arr) {
    if (arr.length < 4) {
      return -1;
    }
    Arrays.sort(arr);
    long maxProduct1 =
        (long) arr[arr.length - 1]
            * arr[arr.length - 2]
            * arr[arr.length - 3]
            * arr[arr.length - 4];
    long maxProduct2 = (long) arr[0] * arr[1] * arr[arr.length - 1] * arr[arr.length - 2];
    return Math.max(maxProduct1, maxProduct2);
  }

  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int tc = scanner.nextInt();
    while (tc-- > 0) {
      int n = scanner.nextInt();
      int[] arr = new int[n];
      for (int i = 0; i < n; i++) arr[i] = scanner.nextInt();
      System.out.println(maxProductOfFour(arr));
    }
    scanner.close();
  }
}

File: Interview Prep/Find the maximum product of 4 numbers in an array /MaxProductOfFour.cpp
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

class MaxProductOfFour {
 public:
  static long long maxProductOfFour(vector<int>& arr) {
    if (arr.size() < 4) {
      return -1;
    }
    sort(arr.begin(), arr.end());
    long long maxProduct1 = (long long)arr[arr.size() - 1] *
                            arr[arr.size() - 2] * arr[arr.size() - 3] *
                            arr[arr.size() - 4];
    long long maxProduct2 =
        (long long)arr[0] * arr[1] * arr[arr.size() - 1] * arr[arr.size() - 2];
    return max(maxProduct1, maxProduct2);
  }
};

int main() {
  int tc;
  cin >> tc;
  while (tc-- > 0) {
    int n;
    cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; ++i) {
      cin >> arr[i];
    }
    cout << MaxProductOfFour::maxProductOfFour(arr) << endl;
  }
  return 0;
}